AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -785,6 +785,33 @@ - (NSMutableDictionary *)getReportingParametersForUpdatedRelease:(NSString *)upd
   return reportingParameters;
 }
 
+- (void)changeDistributionGroupIdAfterAppUpdateIfNeeded {
+  NSString *lastDownloadedReleaseHashes = [MS_USER_DEFAULTS objectForKey:kMSDownloadedReleaseHashKey];
+  NSString *lastDownloadedDistributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDownloadedDistributionGroupIdKey];
+  if ((lastDownloadedDistributionGroupId == nil) || ([self isCurrentReleaseWasUpdated:lastDownloadedReleaseHashes] == NO)) {
+    return;
+  }
+  NSString *currentDistributionGroupId = [MS_USER_DEFAULTS objectForKey:kMSDistributionGroupIdKey];
+  if ((currentDistributionGroupId != nil) && ([lastDownloadedDistributionGroupId isEqualToString:currentDistributionGroupId])) {
+    return;
+  }
+
+  // Set group ID from downloaded release details.
+  MSLogDebug([MSDistribute logTag], @"Current group ID doesn't match the group ID of downloaded release, updating current group id: %@", lastDownloadedDistributionGroupId);
+  [MS_USER_DEFAULTS setObject:lastDownloadedDistributionGroupId forKey:kMSDistributionGroupIdKey];
+
+  // Remove saved downloaded group ID.
+  [MS_USER_DEFAULTS removeObjectForKey:kMSDownloadedDistributionGroupIdKey];
+}
+
+- (bool)isCurrentReleaseWasUpdated:(NSString *)lastDownloadedReleaseHashes {
+  if (lastDownloadedReleaseHashes == nil) {
+    return NO;
+  }
+  NSString *currentInstalledReleaseHash = MSPackageHash();
```

Comment:
We need to check `MSPackageHash` in terms of performance. You removed all currentReleaseHash from methods' parameter that seems like you call this method whenever you need it.